chain/types/ethtypes/eth_transactions.go

```diff
@@ -103,18 +101,18 @@ func EthTxArgsFromMessage(msg *types.Message) (EthTxArgs, error) {
 			if err != nil {
 				return EthTxArgs{}, xerrors.Errorf("failed to read params byte array: %w", err)
 			}
-
-			if len(params) == 0 {
-				// Otherwise, we don't get a guaranteed round-trip.
-				return EthTxArgs{}, xerrors.Errorf("cannot invoke contracts with empty parameters from an eth-account")
-			}
 		}
 	}
 
 	if paramsReader.Len() != 0 {
 		return EthTxArgs{}, xerrors.Errorf("extra data found in params")
 	}
 
+	if len(params) == 0 && msg.Method != builtintypes.MethodSend {
```

Comment:
@Stebalien I _think_ we need this to cover a case where the EthTx calls the EAM's CreateExternal method with the CBOR-encoding of the empty byte array, which would fail to roundtrip.

Reply:
Feels cleaner regardless.

Reply:
Yes, but isn't that case already covered? I.e., we should fail to read a bytes array from the params in that case. But either way works.